pjrt/convert/utils/xplane_visitor.zig

```diff
@@ -0,0 +1,234 @@
+        return self.line.events.len;
+    }
+};
+
+const XEventMetadataInstance = xplane_proto.XEventMetadata.init();
+const XStatMetadataInstance = xplane_proto.XStatMetadata.init();
+
+pub const XPlaneVisitor = struct {
+    plane: *const xplane_proto.XPlane,
+    event_type_by_id: std.AutoHashMapUnmanaged(i64, xplane_schema.HostEventType) = .{},
+    stat_type_by_id: std.AutoHashMapUnmanaged(i64, xplane_schema.StatType) = .{},
+    stat_metadata_by_type: std.AutoHashMapUnmanaged(xplane_schema.StatType, *const xplane_proto.XStatMetadata) = .{},
+
+    const EventTypeGetter = *const fn ([]const u8) ?xplane_schema.HostEventType;
+    const StatTypeGetter = *const fn ([]const u8) ?xplane_schema.StatType;
+
+    pub fn init(
+        allocator: std.mem.Allocator,
+        plane: *const xplane_proto.XPlane,
+        event_type_getter_list: []const EventTypeGetter,
+        stat_type_getter_list: []const StatTypeGetter,
+    ) !XPlaneVisitor {
+        var res: XPlaneVisitor = .{ .plane = plane };
+        try res.buildEventTypeMap(allocator, plane, event_type_getter_list);
+        try res.buildStatTypeMap(allocator, plane, stat_type_getter_list);
+        return res;
+    }
+
+    pub fn buildEventTypeMap(
+        self: *XPlaneVisitor,
+        allocator: std.mem.Allocator,
+        plane: *const xplane_proto.XPlane,
+        event_type_getter_list: []const EventTypeGetter,
+    ) !void {
+        if (event_type_getter_list.len == 0) return;
+        for (plane.event_metadata.items) |event_metadata| {
+            const metadata_id = event_metadata.key;
+            const metadata = event_metadata.value.?;
+            for (event_type_getter_list) |event_type_getter| {
+                if (event_type_getter(metadata.name.getSlice())) |event_type| {
+                    try self.event_type_by_id.put(allocator, metadata_id, event_type);
+                    break;
+                }
+            }
+        }
+    }
+
+    pub fn buildStatTypeMap(
+        self: *XPlaneVisitor,
+        allocator: std.mem.Allocator,
+        plane: *const xplane_proto.XPlane,
+        stat_type_getter_list: []const StatTypeGetter,
+    ) !void {
+        if (stat_type_getter_list.len == 0) return;
+        for (plane.stat_metadata.items) |stat_metadata| {
+            const metadata_id = stat_metadata.key;
+            const metadata = stat_metadata.value.?;
+            for (stat_type_getter_list) |stat_type_getter| {
+                if (stat_type_getter(metadata.name.getSlice())) |stat_type| {
+                    try self.stat_type_by_id.put(allocator, metadata_id, stat_type);
+                    try self.stat_metadata_by_type.put(allocator, stat_type, &stat_metadata.value.?);
+                    break;
+                }
+            }
+        }
+    }
+
+    pub fn getEventType(self: *const XPlaneVisitor, event_metadata_id: i64) ?xplane_schema.HostEventType {
+        return self.event_type_by_id.get(event_metadata_id);
```

Comment:
instead of returning an "?HostEventType" return HostEventType with .unknown for non found values.